pastis/matrix_generation/matrix_building_numerical.py

```diff
@@ -435,6 +464,57 @@ def _hicat_matrix_one_pair(norm, wfe_aber, resDir, savepsfs, saveopds, segment_p
     return contrast, segment_pair
 
 
+def _rst_matrix_one_pair(norm, wfe_aber, resDir, savepsfs, saveopds, actuator_pair):
+    """
+    Function to calculate RST mean contrast of one aberrated segment pair in CGI.
+    :param norm: float, direct PSF normalization factor (peak pixel of direct PSF)
+    :param wfe_aber: calibration aberration per segment in m
+    :param resDir: str, directory for matrix calculations
+    :param savepsfs: bool, if True, all PSFs will be saved to disk individually, as fits files
+    :param saveopds: bool, if True, all pupil surface maps of aberrated segment pairs will be saved to disk as PDF
+    :param actuator_pair:
+    :return: contrast as float, and segment pair as tuple
+    """
+
+    # Set up RST simulator in coronagraphic state
+    rst_cgi = webbpsf_imaging.set_up_cgi()
+
+    # Put aberration on correct segments. If i=j, apply only once!
+    log.info(f'PAIR: {actuator_pair[0]}-{actuator_pair[1]}')
+
+    # Transform segment to actuators coordonates
+    nb_actu = rst_cgi.nbactuator
+    actu_i_x , actu_i_y = util.seg_to_dm_xy(nb_actu, actuator_pair[0])
+    actu_j_x , actu_j_y = util.seg_to_dm_xy(nb_actu, actuator_pair[1])
+
+
+    # Put aberration on correct segments. If i=j, apply only once!
+    rst_cgi.dm1.flatten()
+    rst_cgi.dm1.set_actuator(actu_i_x, actu_i_y, wfe_aber)
+    if actuator_pair[0] != actuator_pair[1]:
+        rst_cgi.dm1.set_actuator(actu_j_x, actu_j_y, wfe_aber)
+
+    log.info('Calculating coro image...')
+    image = rst_cgi.calc_psf(nlambda=1, fov_arcsec=1.6)
+    psf = image[0].data / norm
+
+    # Save PSF image to disk
+    if savepsfs:
+        filename_psf = f'psf_actuator_{actuator_pair[0]}-{actuator_pair[1]}'
+        hcipy.write_fits(psf, os.path.join(resDir, 'psfs', filename_psf + '.fits'))
+
+    # Plot segmented mirror WFE and save to disk
+    if saveopds:
+        opd_name = f'opd_actuator_{actuator_pair[0]}-{actuator_pair[1]}'
+        plt.clf()
+        plt.figure(figsize=(8, 8))
+        rst_cgi.dm1.display(what='opd', opd_vmax=wfe_aber, colorbar_orientation='horizontal', title='Aberrated segment pair')
+        plt.savefig(os.path.join(resDir, 'OTE_images', opd_name + '.pdf'))
+
+    contrast = rst_cgi.raw_contrast()
```

Comment:
Using `raw_contrast()` here is ok, however I would discourage the usage of this function inside of pastis; see my main review comment.